vulkan_generator/main.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This is the entry point for Vulkan Code Generator"""
+
+import sys
+
+from vulkan_gen import generator
+
+
+def main() -> int:
+    """ Entry point """
+    result = generator.generate(sys.argv[1])
+    assert result
+    return 0
```

Comment:
Very minor nits: return value of 0 is unused. Return value of generate is always true. It can be nice to just rely on exceptions at this level (i.e. there is no "sensible" reason for `generate` to fail, so there is no need for a return value from `generate`, nor from `main`, as `generate` will just raise an exception on failure; if the exception is uncaught, the exit code from the script is non-zero anyway - no need to check it and return 1). Alternatively, if `main` _does_ return a value, then perhaps it should be called as such `sys.exit(main())`.

Reply:
This was more of a placeholder but I believe I should actually check the certain error cases already e.g. XML could not found etc.

Reply:
Or we can let the exceptions handle it. Which seems quite descriptive already for our use case.